### DISCUSSION TITLE
Changes to productive

### DISCUSSION THREAD

**Discussion by SandraMarq** [open-sdg/open-sdg#1893] [General]
Hello.

When you push changes to productive, it takes a long time to reflect them.

Is this how it works?

Thank you

**Comment by brockfanning** [comment]
@SandraMarq The build time varies depending on several factors. But how long exactly is it taking? I wouldn't expect it to take longer than 20 minutes or so.

**Comment by SandraEliz** [comment]
Hello

Passing change relative to data takes minutes.
But when I pass changes to metadata or some configuration file, it takes a long time.
The last time it was 1 hour and the changes were not reflected.

I had to make a change to a file in the site repository and push it to production for the changes in the data repository to be reflected.

Thank you

**Comment by otis-bath** [comment]
@SandraEliz - have you set up triggered site builds and automatic rebuilds? https://open-sdg.readthedocs.io/en/latest/automation/triggered-site-builds/ This should then perform a site build when you make a change in the data repository

**Reply by SandraMarq** [reply]
Hello
I already have them generated

**Comment by SandraMarq** [comment]
Hello.

Again I changed to the productive site and they are not reflected.

I waited 1 hour and nothing, in the end I made changes in the site repository and sent them to the production site so that the changes could be seen.

What will I be missing?

Thank you

**Comment by brockfanning** [comment]
@SandraMarq I believe that to do this, you would need to do the following:

1. Add two lines to your site repository's `deploy-to-production.yml` file. These lines should be similar to what is in your `deploy-to-staging.yml` file, but a bit different, under the `on:` section:

        repository_dispatch:
          types: rebuild-production

2. Then you would need to add some lines to your data repository's `deploy-to-production.yml` file. These lines should be similar to what is in your `deploy-to-staging.yml` file, but a bit different, at the end of the `steps:` section:

        - name: Waiting 4 minutes for production data deployment
          run: |
            sleep 240
        - name: Rebuild site
          run: |
            curl -X POST https://api.github.com/repos/sdg-odsmexico/site/dispatches \
            -H "Accept: application/vnd.github.everest-preview+json" \
            -H "Authorization: token ${{secrets.token}}" \
            --data '{"event_type": "rebuild-production"}'

**Comment by SandraEliz** [comment]
Hello.

I have already made the changes in the files that you mention, but when I pass changes to productive they are still not reflected.

Yesterday I passed them and I reviewed it today and the changes are not reflected.

**Reply by brockfanning** [reply]
@SandraEliz What is the specific change you're expecting to see? I can check also to see whether it might be browser caching.

**Comment by SandraEliz** [comment]
I made a change in the objective text in the metadata, from indicator 1.1.1

In pre-production it looks like this:
![image](https://user-images.githubusercontent.com/117297044/207902203-b1456999-1198-4c07-b074-8f3070835ce3.png)

And in production like this:
![image](https://user-images.githubusercontent.com/117297044/207902575-27058c27-0234-477f-8b7e-a5fb66ec72a2.png)

The productive link is:  https://prod-odsmexico.github.io/1-1-1/

**Comment by brockfanning** [comment]
@SandraEliz Try changing [this line](https://github.com/prep-odsmexico/site/blob/develop/.github/workflows/deploy-to-production.yml#L25) to "rebuild-production" instead of "rebuild-staging".

**Comment by SandraEliz** [comment]
I already made the change and the changes are still not reflected in production

**Comment by brockfanning** [comment]
@SandraEliz I'm seeing "cambio2" now. If you want to confirm again, maybe change it to "cambio3" and then do another production release on the data repository. It should trigger a production release on the site too.

**Comment by SandraEliz** [comment]
I made that change "cambio3" and it is not reflected

**Reply by brockfanning** [reply]
@SandraEliz This line is incorrect: https://github.com/prep-odsmexico/data/blob/75abf62f8f56f610d588018f96162ac3f25cb0d2/.github/workflows/deploy-to-production.yml#L62

It should be pointing to your staging site repository. In other words, it needs to be identical to this line: https://github.com/prep-odsmexico/data/blob/develop/.github/workflows/deploy-to-staging.yml#L55

**Reply by SandraMarq** [reply]
Hello

It works now.

Thanks
